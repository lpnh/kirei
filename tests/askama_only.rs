use kirei::AskamaFormatter;

fn create_formatter() -> AskamaFormatter {
    AskamaFormatter::new().expect("Failed to create formatter")
}

#[test]
fn comment() {
    let mut formatter = create_formatter();
    let input = r#"{# A Comment #}"#;

    let formatted_output = formatter.format(input).expect("Formatting failed");

    insta::assert_snapshot!(input);
    insta::assert_snapshot!(formatted_output);
}

#[test]
fn empty_if() {
    let mut formatter = create_formatter();
    let input = r#"{% if foo %}{% endif %}"#;

    let formatted_output = formatter.format(input).expect("Formatting failed");

    insta::assert_snapshot!(input);
    insta::assert_snapshot!(formatted_output);
}

#[test]
fn if_with_text() {
    let mut formatter = create_formatter();
    let input = r#"{% if foo %} this is some text {% endif %}"#;

    let formatted_output = formatter.format(input).expect("Formatting failed");

    insta::assert_snapshot!(input);
    insta::assert_snapshot!(formatted_output);
}

#[test]
fn simple() {
    let mut formatter = create_formatter();
    let input = r#"{{ user.name }}{% for item in items %}{{ item }}{% endfor %}{# comment here #}"#;

    let formatted_output = formatter.format(input).expect("Formatting failed");

    insta::assert_snapshot!(input);
    insta::assert_snapshot!(formatted_output);
}
