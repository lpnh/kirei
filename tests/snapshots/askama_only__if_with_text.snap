---
source: tests/askama_only.rs
expression: input
---
{% if foo %} this is some text {% endif %}
